modules/templates/templates_plugin.rb

```diff
@@ -6,9 +6,5 @@ class Plugin < ::Proxy::Plugin
     plugin :templates, ::Proxy::VERSION
 
     validate_presence :template_url
-
-    after_activation do
-      loading_failed "missing :foreman_url: from configuration." unless Proxy::SETTINGS.foreman_url
```

Comment:
This is actually required. TemplatesAPI creates a [TemplateProxyRequest](https://github.com/theforeman/smart-proxy/blob/develop/modules/templates/templates_api.rb#L19), which calls [`request_factory`](https://github.com/theforeman/smart-proxy/blob/develop/modules/templates/template_proxy_request.rb#L26) which [requires this setting](https://github.com/theforeman/smart-proxy/blob/develop/lib/proxy/request.rb#L57).

Reply:
Ah, that does explain it.